### PR TITLE
Add check for empty theme URI before saving theme settings

### LINKED ISSUE #2850: BTCPay looks weird if you save theme settings with empty theme value
**Describe the bug**

See this screenshot:

Screen Shot 2021-09-06 at 5 21 01 PM

**To Reproduce the bug**
Steps to reproduce the reported bug:
1. Go to Server Settings tab
2. Go to Theme tab in the left sidebar
3. Click on "Use custom theme" toggle to enable it
4. Click on "Save" button
5. Notice that the page looks messed up

**Expected behavior**

Expect to receive an error message telling me something is wrong instead of applying an empty theme value.

**Your BTCPay Environment (please complete the following information):**
- BTCPay Server Version: v1.2.2.0
- Deployment Method: Doesn't matter
- Browser: Doesn't matter

### PR DESCRIPTION
**Pull request by bolatovumar**
|Before|After|
|---|---|
|Screen Shot 2021-09-06 at 5 21 01 PM|Screen Shot 2021-09-06 at 8 26 12 PM|

close #2850

### CODE REVIEW

File: BTCPayServer/Controllers/ServerController.cs [hunk 1]
```diff
@@ -979,13 +976,21 @@ public async Task<IActionResult> Theme()
...
         }
-        
+
         [Route("server/theme")]
         [HttpPost]
         public async Task<IActionResult> Theme(ThemeSettings settings)
         {
-            await _SettingsRepository.UpdateSetting(settings);
-            TempData[WellKnownTempData.SuccessMessage] = "Theme settings updated successfully";
+            if (settings.CustomTheme && !Uri.IsWellFormedUriString(settings.CssUri, UriKind.RelativeOrAbsolute))
```
Comment (bolatovumar) [inline BTCPayServer/Controllers/ServerController.cs:984]: This is the real change, everything else is me adding indicators that things can be `null`.

File: BTCPayServer/Controllers/ServerController.cs [hunk 2]
```diff
@@ -1076,6 +1081,7 @@ public async Task<IActionResult> LogsView(string file = null, int offset = 0)
 
                 var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(_Options.LogFile);
                 var fileExtension = Path.GetExtension(_Options.LogFile) ?? string.Empty;
+                // We are checking if "di" is null above yet accessing GetFiles on it, this could lead to an exception?
```
Comment (bolatovumar) [inline BTCPayServer/Controllers/ServerController.cs:1084]: Added this comment to bring attention to this potential issue. Seems like it _could_ result in a bug?

File: BTCPayServer/Controllers/ServerController.cs [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#nullable enable
```
Comment (Kukks) [inline BTCPayServer/Controllers/ServerController.cs:1]: You have a real fetish for this @bolatovumar

File: BTCPayServer/Controllers/ServerController.cs (hunk 2 above)
Comment (Kukks) [reply BTCPayServer/Controllers/ServerController.cs:1084]: Indeed it is a bug

**Review by Kukks — APPROVED** [review]

**Pull request merged**